src/main/detectors/base-detector.ts

```diff
@@ -0,0 +1,37 @@
+import request, { RequestPromiseOptions } from "request-promise-native"
+import { Cluster } from "../cluster";
+
+export class BaseDetector {
+  cluster: Cluster
+  accuracy = 0
+  key: string
+  value: string|number|boolean
+  success: boolean
+
+  constructor(cluster: Cluster) {
+    this.cluster = cluster
+  }
+
+  detect(): Promise<void> {
```

Comment:
Just wondering why `detect` does not return an object with value/accuracy?

Reply:
Refactored this and now `detect` returns `{ value: <value>, accuracy: <accuracy>}`